opendmarc/opendmarc.c

```diff
@@ -2452,8 +2563,54 @@ mlfi_eom(SMFICTX *ctx)
 			}
 			else if (ar.ares_result[c].result_method == ARES_METHOD_ARC)
 			{
-				if (ar.ares_result[c].result_result == ARES_RESULT_PASS)
-					dfc->mctx_arcpass = TRUE;
+				limit_arc++;
+				/*
+				** If we already came across a trusted A-R header with arc=pass
+				** we need to fail.
+				*/
+				if (ar.ares_result[c].result_result == ARES_RESULT_PASS && limit_arc == 1)
+				{
+					dfc->mctx_arcpass = ARES_RESULT_PASS;
+				}
+				else
+				{
+					dfc->mctx_arcpass = ARES_RESULT_FAIL;
+				}
+
+				/*
+				** Check arc status against whitelist policy
```

Comment:
This could be a single-line comment; block comments are typically followed by a blank line.